src/libs/conduit/conduit_node.cpp

```diff
@@ -13830,8 +13830,8 @@ Node::diff_compatible(const Node &n, Node &info, const float64 epsilon) const
         {
             // NOTE: Can't use 'value' for characters since type aliasing can
             // confuse the 'char' type on various platforms.
-            char_array t_array((const void*)as_char8_str(), dtype());
-            char_array n_array((const void*)n.as_char8_str(), n.dtype());
+            char_array t_array((const void*)m_data, dtype());
+            char_array n_array((const void*)n.m_data, n.dtype());
             res |= t_array.diff_compatible(n_array, info, epsilon);
```

Comment:
got it: 
as_char8_str() already applies the offset, by then using the dtype again ends up double applying.